core/src/main/java/org/neo4j/ogm/metadata/FieldInfo.java

```diff
@@ -289,15 +278,15 @@ public boolean isLabelField() {
     }
 
     public boolean isArray() {
-        return descriptor.startsWith("[");
+        return descriptor.endsWith("[]");
```

Comment:
No. bytecode parsing representations vs string representations of classes are very different. This is only the first (and major) step in the refactor. Over time you would remove all the string comparison stuff because you can either interrogate the class file or org.reflections.